sycamore/transforms/embed.py

```diff
@@ -85,7 +92,69 @@ def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
         text_batch = ["" if doc.text_representation is None else doc.text_representation for doc in doc_batch]
         embeddings = self._transformer.encode(text_batch, batch_size=self.model_batch_size, device=self.device)
         for doc, embedding in zip(doc_batch, embeddings):
-            doc.embedding = embedding
+            doc.embedding = embedding.tolist()
+
+        return doc_batch
+
+
+class OpenAIEmbeddingModels(Enum):
+    TEXT_EMBEDDING_ADA_002 = "text-embedding-ada-002"
+
+
+class OpenAIEmbedder(Embedder):
+    """Embedder implementation using the OpenAI embedding API.
+
+    Args:
+        model_name: The name of the OpenAI embedding model to use.
+        batch_size: The Ray batch size.
+        model_batch_size: The number of documents to send in a single OpenAI request.
+    """
+
+    def __init__(
+        self,
+        model_name: str = OpenAIEmbeddingModels.TEXT_EMBEDDING_ADA_002.value,
+        batch_size: Optional[int] = None,
+        model_batch_size: int = 100,
+        api_key: Optional[str] = None,
+    ):
+        super().__init__(model_name, batch_size, model_batch_size, device="cpu")
+
+        if api_key is None:
+            api_key = os.environ.get("OPENAI_API_KEY", None)
+
+        assert api_key is not None, (
+            "You must provide an API key to "
+            "use the LLM. Either pass it in "
+            "the constructor or set the "
+            "OPENAI_API_KEY environment "
+            "variable."
+        )
+        self.api_key = api_key
+        self.model_name = model_name
```

Comment:
Does it make sense to re-use the OpenAI class here? [Link](https://github.com/aryn-ai/sycamore/blob/main/sycamore/llms/llms.py#L31)

Reply:
I wasn't sure. On one hand, there is some common configuration, but on the other hand embedding models are different and have a different signature. We could have the LLM class have an embed method, but then we would need some options for LLMs that don't support embeddings, etc. I'm open to suggestions, but it sort of seemed like using the OpenAI LLM class would add more code than it saves.